didman/didman.go

```diff
@@ -223,6 +252,19 @@ func (d *didman) validateCompoundServiceEndpoint(references map[string]ssi.URI)
 	return nil
 }
 
+func filterCompoundServices(doc *did.Document) []did.Service {
+	var compoundServices []did.Service
+	for _, service := range doc.Service {
+		if service.Type == ContactInformationServiceType {
+			continue
+		}
+		if _, ok := service.ServiceEndpoint.(map[string]interface{}); ok {
```

Comment:
I think this is rather naive. Don't we have a more concise way to determine whether an endpoint is (_probably_) a compound service?

Reply:
The `ServiceEndpoint` of `Endpoint` and `EndpointRef` are of type `string`, compound is type `map[string]interface{}`. I don't really see another way?